mod/item.php

```diff
@@ -369,58 +369,53 @@ function item_post(App $a) {
 
 	// Look for any tags and linkify them
 	$inform   = '';
-
-	$tags = BBCode::getTags($body);
-
-	$tagged = [];
-
 	$private_forum = false;
+	$private_id = null;
 	$only_to_forum = false;
 	$forum_contact = [];
 
-	if (count($tags)) {
+	BBCode::performWithEscapedTags($body, ['noparse', 'pre', 'code'], function ($body) use ($profile_uid, $network, $str_contact_allow, &$inform, &$private_forum, &$private_id, &$only_to_forum, &$forum_contact) {
```

Comment:
I would prefer having this extracted in a separate function. On the first glance one can easily miss that this is a function inside of a function where some variables are shared with the outside world. This can create a lot of confusion.

Reply:
I agree. I have tried to minimize the changes to the existing code but it really isn't ideal.